x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/fleet_server_upgrade_modal.tsx

```diff
@@ -0,0 +1,169 @@
+
+  return (
+    <EuiModal onClose={onClose}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <FormattedMessage
+            id="xpack.fleet.fleetServerUpgradeModal.modalTitle"
+            defaultMessage="This version of Fleet requires a Fleet Server"
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiImage
+          src={getAssetsPath('./announcement.jpg')}
+          alt={i18n.translate('xpack.fleet.fleetServerUpgradeModal.announcementImg', {
+            defaultMessage: 'Announcement Image',
+          })}
+        />
+        <EuiSpacer size="m" />
+        <EuiText>
+          {isCloud ? (
+            <FormattedMessage
+              id="xpack.fleet.fleetServerUpgradeModal.textPart1Cloud"
+              defaultMessage="Fleet Server is now available and it provides improved scalability and security. If you already had APM on Elastic Cloud, we've upgraded it to APM & Fleet. {strong} To continue using Fleet, you must install a Fleet Server and the new version of Elastic Agent on each host. Learn more in our {link}."
+              values={{
+                strong: (
+                  <strong>
+                    <FormattedMessage
+                      id="xpack.fleet.fleetServerUpgradeModal.existingAgentText"
+                      defaultMessage="Your existing Elastic Agents have been automatically unenrolled and have stopped sending data."
+                    />
+                  </strong>
+                ),
+                link: (
+                  <EuiLink
+                    href="https://www.elastic.co/guide/en/fleet/current/upgrade-elastic-agent.html"
+                    external={true}
+                  >
+                    <FormattedMessage
+                      id="xpack.fleet.fleetServerUpgradeModal.fleetServerMigrationGuide"
+                      defaultMessage="Fleet Server migration guide"
+                    />
+                  </EuiLink>
+                ),
+              }}
+            />
+          ) : (
+            <FormattedMessage
+              id="xpack.fleet.fleetServerUpgradeModal.textPart1OnPremise"
+              defaultMessage="Fleet Server is now available and it provides improved scalability and security. {strong} To continue using Fleet, you must install a Fleet Server and the new version of Elastic Agent on each host. Learn more in our {link}."
+              values={{
+                strong: (
+                  <strong>
+                    <FormattedMessage
+                      id="xpack.fleet.fleetServerUpgradeModal.existingAgentText"
+                      defaultMessage="Your existing Elastic Agents have been automatically unenrolled and have stopped sending data."
+                    />
+                  </strong>
+                ),
+                link: (
+                  <EuiLink
+                    href="https://www.elastic.co/guide/en/fleet/current/upgrade-elastic-agent.html"
```

Comment:
is this the right set of docs to point to? should we use the Fleet user guide short URL that we display in the empty state?